users/models.py

```diff
@@ -81,15 +91,18 @@ class Profile(models.Model):
     post_karma = models.IntegerField(default=0)
     gold_awards = models.IntegerField(default=0)
     gender = models.CharField(choices=GENDER_CHOICES, max_length=1)
-    avatar = models.ImageField(upload_to="users_avatars/", null=True, blank=True, default=None)
+    avatar = models.ImageField(upload_to=user_avatar_path, null=True, blank=True, default=None)
+    banner = models.ImageField(upload_to=user_banner_path, null=True, blank=True, default=None)
     user = models.OneToOneField("User", on_delete=models.CASCADE, null=False)
 
     def __str__(self: "Profile") -> str:
         return f"{self.user.nickname}"
 
     def save(self: "Profile", *args: any, **kwargs: dict) -> None:
         if self.avatar:
-            self.avatar = self.process_avatar(self.avatar)
+            self.avatar = self.process_image(self.avatar, (32, 32))
```

Comment:
hmm we might have a bug here. And it was in original implementation. Simply, it was overlooked. 
The file is processed every time when self.avatar or self.banner is set. So, even if the image is not uploaded the image is processed. 
I think, I would use __init__ method to store self._initial_banner and self._initial_avatar to store initial values of banner and avatar, and then check if the values were changed before calling process image in save.

Reply:
Corrected :>